Added external EntraId documentation

sebgott opened the pull request:
# Description
To help external users understand how Event Issuer will support Entra Id. We added a documentation page on how the authZ/authN flow works with background services using EntraId and how we will support it.

# Impacts assessment
- New wiki page on subscription authentication

# Reviewer checklist

- [x] Title of PR is relevant and descriptional.
- [x] Code changes are relevant and clear. If any hardcode is created necessary comments has been added.
- [x] User Story or Bug has been created if there are workarounds to issues at other libraries.
- [x] Pull Request description contains relevant information that matches the scope of the pull request.
- [ ] Sufficient impact assesment is written.
- [x] Documentation example README or wiki is up to date with code.

DO NOT DELETE:
[+semver:minor]

sebgott committed:
Added external EntraId documentation

schteffy commented on docs/authentication/subscription-authentication.md:
````diff
@@ -0,0 +1,37 @@
+This diagram describes how authentication works between background services.
+
+For Event-Issuer the flow would look like this
+``` mermaid
````
As the Admin app registration creation is not relevant in this case, can be included as a pre-condition instead in the diagram itself.

sebgott replied:
Removed the app registration creation from the sequence diagram and added it as a pre-condition for the process.

sebgott committed:
Modified sequence diagram with pre condition

idarb-oss committed:
Update subscriptoin with auth information.

Needs additional information for the different parts with how to configure. To be added as the development continues

sebgott committed:
Update for wiki

idarb-oss committed:
Add produce and update subscribe documentation. Add the new open api specification

idarb-oss committed:
Use redocly instead for API View
